screensharing/main.js

```diff
@@ -0,0 +1,113 @@
+const { app, BrowserWindow, ipcMain, screen } = require('electron');
+
+const { SCREEN_SHARE_EVENTS_CHANNEL, SCREEN_SHARE_EVENTS, TRACKER_SIZE } = require('./constants');
+
+/**
+ * Main process component that sets up electron specific screen sharing functionality, like screen sharing
+ * tracker and window selection.
+ * The class will process events from {@link ScreenShareRenderHook} initialized in the renderer, and the
+ * always on top screen sharing tracker window.
+ */
+class ScreenShareMainHook {
+    /**
+     * Create ScreenShareMainHook linked to jitsiMeetWindow.
+     *
+     * @param {BrowserWindow} jitsiMeetWindow - BrowserWindow where jitsi-meet api is loaded.
+     * @param {string} identity - Name of the application doing screen sharing, will be displayed in the
+     * screen sharing tracker window text i.e. {identity} is sharing your screen.
+     */
+    constructor(jitsiMeetWindow, identity) {
+        this._jitsiMeetWindow = jitsiMeetWindow;
+        this._identity = identity;
+        this._onScreenSharingEvent = this._onScreenSharingEvent.bind(this);
+
+        // Listen for events coming in from the main render window and the screen share tracker window.
+        ipcMain.on(SCREEN_SHARE_EVENTS_CHANNEL, this._onScreenSharingEvent);
+
+        // Clean up ipcMain handlers to avoid leaks.
+        app.once('window-all-closed', () => {
```

Comment:
I guess even if only `jitsiMeetWindow` is closed we can remove the listener, isn't it? If another one is opened the consumer of the API will anyway need to call all setup methods in order for the screen sharing to work.

Also do we close the screen share tracker if `jitsiMeetWindow` is closed. I think we do because we will receive `_willDispose` event from the iframe api, but can you confirm please?

Reply:
Indeed it does make more sense to remove the listener when jitsiMeetWindow is closed, will do that.
Yes the screen share tracker is removed when jitsiMeetWindow is closed due to the conference ending event or the _willDispose event from the iframe.